src/main/scala/com/microsoft/hyperspace/index/IndexCacheFactory.scala

```diff
@@ -16,6 +16,7 @@
 
 package com.microsoft.hyperspace.index
 
+import org.apache.hadoop.yarn.util.SystemClock
```

Comment:
Is this OK to use? I see the following from https://hadoop.apache.org/docs/stable2/api/org/apache/hadoop/yarn/util/SystemClock.html#getTime(): 
`NOTE: Do not use this to calculate a duration of expire or interval to sleep`

Reply:
SystemClock in hadoop internally implementation uses `System.currentTimeMillis()` which is what was used before.

But also what it says is correct, an admin can come and change time of machine which could give wrong computation for our caching. We should be running internal clocks for caching, and can actually do that as a separate change ?

Reply:
> SystemClock in hadoop internally implementation uses `System.currentTimeMillis()` which is what was used before.

Ah OK. In that case we can just leave it as it is.

Reply:
Can use something like this https://guava.dev/releases/21.0/api/docs/com/google/common/base/Ticker.html

Reply:
We can just use `MonotonicClock` as the above doc suggests: https://hadoop.apache.org/docs/stable2/api/org/apache/hadoop/yarn/util/MonotonicClock.html.

Reply:
Ah, MonotonicClock only comes into hadoop utils from 2.8 and we are using 2.65 package, can I update it to 2.8, I don't know how much backward compatibility hyperspace can handle with hadoop git update ?

Reply:
For now, let's just leave it as `SystemClock`.